weather.py

```diff
@@ -90,7 +139,55 @@ def generate_summary(weather_data):
     Returns:
         A string containing the summary information.
     """
-    pass
+    
+#5 Day Overview
+      # The lowest temperature will be 9.4°C, and will occur on Friday 02 July 2021.
+      # The highest temperature will be 20.0°C, and will occur on Saturday 03 July 2021.
+      # The average low this week is 12.2°C.
+      # The average high this week is 17.8°C.
+
+    #now we need to calculate each element in the output string and insert them in
+    num_days = len(weather_data)
+
+    #create arrays with the min_temp and max_temps
+    min_temps = []
+    max_temps = []
+    for element in weather_data:
+        min_temps.append(element[1])
+        max_temps.append(element[2])
+
+    #use the min temp function to get the min_temp and index
+    low_temp_day = find_min(min_temps)
```

Comment:
Can also use tuple unpacking here, since your `find_min` function returns a tuple e.g.

```suggestion
    low_temp_day, low_index = find_min(min_temps)
```